### ISSUE TITLE
Theme Options translation conundrum

### ISSUE THREAD

**Issue by edbloom** [ReduxFramework/redux-framework#1974]
I have all strings in my theme options config wrapped in my themes textdomain to allow users to translate theme options. This worked fine when Redux was embedded in the theme.

I've recently stopped embedding Redux to keep the the theme as light as possible and to ensure users always had the most recent stable version. The issue now is that my strings are not being picked up. I'm guessing this is due to the fact that the plugin is perhaps being loaded before the textdomain is loaded in the theme?

What is the correct way to ensure that my theme option strings can be translated when Redux is not embedded in a theme?

Thanks!

**Comment by kprovance** [comment]
Redux handles it's own string translation.  When you moved to the plugin, it's exactly what it did, no longer relying on your theme's translation matrix.

**Comment by edbloom** [comment]
Hi Kevin,

Thanks for the update. As theme developers what would you then recommend we do? 

My impression is that it would be preferable to have Redux installed as a separate plugin rather than being embedded correct?

But if that's the case, even if I provide a .po/.mo file for users to upload to the plugin folder  it won't be able to hook into the options config that's within my theme (as far I can determine). 

In summary, at present it would then appear that it's impossible for theme authors (and end users) to localize theme options in their language if a theme author elects to have Redux installed as a plugin rather than being embedded?

**Comment by kprovance** [comment]
@Dovy may know.  He's more learned at this whole translation matrix thing than I.  Just an area I never got into.  :\

He'll be along soon, I suspect.  :)

**Comment by edbloom** [comment]
Cheers Kevin :)

**Comment by dovy** [comment]
This would probably be the order in which YOU call the files. Here's why. Your version of "redux" doesn't load until you do. Sure the Redux Framework is included, but your actual fields are included once you run your config. Just make sure your code for translation is running before you setup the Redux Framework object.  ;)

We have a WPML doc in creation as we speak by one of our community members. We'll let you know when it's complete.

**Comment by edbloom** [comment]
Thanks @dovy 

That solved it!
I was loading my textdomain inside a theme setup function attached to after_setup_theme whereas my Redux config was just being required in my theme functions file (below the setup function which is what threw me initially - but your comment made me realize that the Redux config must be running before after_setup_theme is called.

Will be keeping an eye out for the WPML doc - thanks. I'll also definitely be supporting this project in 2015 by way of donations as soon as I can.

**Comment by dovy** [comment]
Thank you. Glad we solved it.

**Comment by edbloom** [comment]
@dovy - just some further follow up on this issue.

My initial solve for this issue hasn't fully worked. I have some Redux options values which are called within some /woocommerce template overrides which stop working once I move my Redux admin-init.php call inside an after_setup_theme function.

My only solve at this point is to take the load_theme_textdomain call out of the after_setup_theme action which at first glance appears to be working ok - but every piece of literature I could find states that load_theme_textdomain should always be called inside after_setup_theme.

**Comment by dovy** [comment]
The issue is simply you're running your config BEFORE after_setup_theme. So just delay the load and it will be resolved.

**Comment by edbloom** [comment]
@dovy - thanks for the super fast response. I'm not sure I follow though.

I've tried the following:
```
function commercegurus_setup() {
	load_theme_textdomain( 'commercegurus', get_template_directory() . '/languages' );
        require get_template_directory() . '/admin/admin-init.php'; // loads options init
}
add_action( 'after_setup_theme', 'commercegurus_setup' );
```
This resolved my original issue, but threw up the issue I reported yesterday. i.e. While the translation of Redux strings is no longer an issue, when I initalize my config as per above, I cannot access config data values in WooCommerce template overrides (i.e. within /mytheme/woocommerce/my-wc-override.php

Is the config now being loaded too late that WooCommerce template overrides don't recognize Redux? 

Do you know of a way of calling the Redux config late enough that the text domain is set first but early enough that WooCommerce overrides recognize Redux option values?

**Comment by edbloom** [comment]
@dovy - any further thoughts on this one? appreciate any input you might have - thanks.
